.github/workflows/android-master.yml

```diff
@@ -0,0 +1,58 @@
+# A workflow for controlling the acions taken by CI whenever we merge code into master
+
+name: Android SDK Master CI
+
+# Triggers the workflow on when code is merged into master
+on:
+  push:
+    branches:
+      - 'master'
+
+jobs:
+  # This workflow contains a single job for running our tests
+  test:
+    name: Run Unit Tests
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so the job can access it
+      - uses: actions/checkout@v1
+
+      # Sets up a JDK on the job so that we can run Java code
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      # Call Gradle to run the unit tests
+      - name: Unit Tests
+        run: bash ./gradlew test --stacktrace
+
+    aar:
+      name: Generate AAR
+      # The type of runner that the job will run on
+      runs-on: ubuntu-latest
+
+      steps:
+        # Checks-out your repository under $GITHUB_WORKSPACE, so the job can access it
+        - uses: actions/checkout@v1
+
+        # Sets up a JDK on the job so that we can run Java code
+        - name: Set up JDK 11
+          uses: actions/setup-java@v1
+          with:
+              java-version: 11
```

Comment:
possibly not a problem but not too sure how well android build tools plays with java 11, i remember have issues with proper linking of classes but maybe that's changed. just pointing out we might end up changing this to 8

Reply:
Uuh good point. Android's Java support should be more up-to-date than it was before. Their support of JDK versions slowed down to push resources into the adoption of Kotlin I heard and with Java 8's end of life approaching at the end of 2020 I would really hope Google updated their support for higher JDK versions by now

Reply:
ah interesting, yeah we should definitely keep an eye out for java 11 support since thats the current LTS but to get going without issues Java 8 is the way to go